defaults/main.yml

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: MIT
+---
+# Here is the right place to put the role's input variables.
+# This file also serves as a documentation for such a variables.
+
+# The provider for nbde_client role. Currently 'clevis' is supported.
+nbde_client_provider: clevis
+
+
+# nbde_client_bindings - the description of the devices, that may also contain
+# an auth configuration for each group of devices.
+#
+# nbde_client_bindings supports the following keys:
+#  * devices - specifies LUKS devices, with their path and possibly either a
+#              passphrase or key file.
+#  * auth    - the settings for the binding, like the servers it should bind
+#              to.
+#  * state   - specifies whether the binding with the configuration described
+#              should be added or removed. Setting state to present (the
+#              default) means bindings will be added; setting state to absent
+#              means bindings will be removed from the devices/slots.
+#
+#
+# devices supports the following keys:
+# * path    - specifies the path to access this device (REQUIRED).
+# * pass    - specifies a valid passphrase for this device.
+# * keyfile - specifies either the absolute or relative path of a valid key
+#             file for this device.
+#
+# Example:
+# - devices:
+#    - path: /dev/sda1
+#      pass: password
+#    - path: /dev/sda2
+#      keyfile: /vault/keyfile
+#
+#
+# auth supports the following keys:
+# * slot               - specifies the slot to use for the binding. Default
+#                        is 1.
+# * servers            - specifies a list of servers to bind to. To enable
+#                        high availability, simply specify more than one server
+#                        here.
+# * threshold          - specifies the threshold for the Shamir Secret Sharing
+#                        (SSS) scheme that is put in place when using more
+#                        than one server. When using multiple servers,
+#                        threshold indicates how many of those servers should
+#                        succeed in order to complete the decryption.
+# * overwrite          - specifies whether we should override any existing
```

Comment:
Note that this is usually the default in Ansible - if you specify a value, it replaces any existing value.  Is this a case where it might be potentially very destructive to overwrite an existing binding?  e.g. there is an irreplaceable private key in a slot?

Reply:
It should not be destructive, since we would be replacing an existing binding with another valid binding, meaning we would be able to unlock the disk with it. We can set this to `yes` as default, then.

Reply:
I think this is the last issue - other than this, looking good

Reply:
Thanks. This was updated, and I am now using `nbde_server` role in testing

Reply:
what happens if the value is set to `no` ?

Reply:
It will not perform the action, if there is already a binding (output with `-v`):
```
TASK [linux-system-roles.nbde_client : Perform clevis operations] ****************************************************************************
ok: [rhel8] => {"changed": false, "msg": "/dev/rhel/root:1 is bound and no overwrite set", "original_bindings": [{"auth": {"discard_passphrase": false, "overwrite": false, "servers": ["localhost:3000"], "slot": 1, "threshold": 1}, "devices": [{"keyfile": "/root/.local/share/nbde_client/keyfile", "path": "/dev/rhel/root"}], "state": "present"}]}
```

Reply:
And it won't report an error? So the role does not do what it was told to do and pretends that everything is ok? To me this seems unusual in the context of Ansible.

Reply:
Yeah, it seems like a bug. I will get this fixed.

Reply:
Is there an actual need for this setting? To me it seems that it could be easier to avoid aving it.

Reply:
has this been solved?

Reply:
Yes, it's gone. Sorry, I forgot to comment here. No need to have this and it was removed.